src/dune_stats/dune_stats_stubs.c

```diff
@@ -0,0 +1,41 @@
+#include <caml/fail.h>
+#include <caml/memory.h>
+#include <caml/mlvalues.h>
+
+#if defined(__APPLE__)
+#include <libproc.h>
+
+CAMLprim value dune_stats_open_fds(value v_pid) {
+  CAMLparam1(v_pid);
+  pid_t pid = Int_val(v_pid);
+  int size = proc_pidinfo(pid, PROC_PIDLISTFDS, 0, NULL, 0);
+  if (size < 0) {
+    caml_failwith("proc_pidinfo failed");
+  }
+  struct proc_fdinfo *fdinfo = (struct proc_fdinfo *)malloc(size);
```

Comment:
if we're just using `size` to get the number of fds, can we skip the second call + the allocation?

Reply:
ok, interesting. I found some more info [in this article](https://zameermanji.com/blog/2021/8/1/counting-open-file-descriptors-on-macos/). My impression is that the 20 extra elements could allow for some flexibility if the number of open FDs change between the 2 calls. It's possible that the data does not fit in the buffer and that it will still return `ENOMEM`, but that's unlikely with the extra 20 slots.
So yes this seems to be the best way to do this (we could allocate `fdinfo` on the stack I suppose)

Reply:
I can try allocating fdinfo on the stack and then just give up on counting anything past a hard limit. Let me try doing that.

Reply:
Alright, I switched to stack allocation.